src/Meilisearch/Index.cs

```diff
@@ -158,8 +168,8 @@ public async Task<UpdateStatus> AddDocuments<T>(IEnumerable<T> documents, string
                 uri = $"{uri}?{new { primaryKey = primaryKey }.ToQueryString()}";
             }
 
-            responseMessage = await this.http.PostJsonCustomAsync(uri, documents);
-            return await responseMessage.Content.ReadFromJsonAsync<UpdateStatus>();
+            responseMessage = await this.http.PostJsonCustomAsync(uri, documents, cancellationToken: cancellationToken).ConfigureAwait(false);
```

Comment:
We have two awaited calls the outer one `this.http.PostJsonCustomAsync` and the inner one inside `PostJsonCustomAsync`.

Each time we do `async` / `await` we create a state machine, `ConfigureAwait(false)` is just here to say when the awaited call is done it can use any available thread in the thread pool instead of waiting for the original thread.

We call it for both just to avoid the extra cost of waiting for the original thread to become available.

Reply:
Ok! Thanks for the explanation!